core/src/main/scala/org/apache/spark/storage/memory/MemoryStore.scala

```diff
@@ -388,7 +388,13 @@ private[spark] class MemoryStore(
     // perform one final call to attempt to allocate additional memory if necessary.
     if (keepUnrolling) {
```

Comment:
shall we merge this `if` into the next `if`?

Reply:
We maybe need request the extra-memory (`bbos.size - unrollMemoryUsedByThisBlock`). So the `keepUnrolling` maybe change in line 393.